proto/provenance_verification.proto

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package verification.provenance.v1;
+
+option go_package = "proto/verification";
+
+message VerificationOptions {
+    oneof option {
+        SkipVerification skip_provenance_verification = 1;
+        ProvenanceReferenceValues reference_provenance = 2;
+    }
+}
+
+// A singleton value to indicate that verification can be skipped.
+message SkipVerification{}
+
+// A collection of reference values for verifying provenance statements.
+message ProvenanceReferenceValues {
+    bool must_have_build_command = 1;
+    Digests reference_builder_image_digests = 3;
+    string reference_repo_uri = 4;
+    StringAllowList reference_builders = 5;
+}
+
+// An allow list of digests, represented as a mapping from cryptographic hash
+// function names, to their allow listed values. 
+message Digests {
+    // Keys are cryptographic hash algorithms (e.g., sha256).
```

Comment:
I thought we were using the numeric multihash id somewhere else (from https://github.com/multiformats/multicodec/blob/master/table.csv ).

in particular `sha256` is ambiguous; it should be `sha2-256` or something like that, to be unambiguos -- but the numeric id solves that problem (without having to actual implement full multihash -- I don't think we should do that for now).

Reply:
Not in this repo. This repo follows the in-toto and SLSA style of using digests. We can change that in the future, but that is out of the scope of this PR.